skrub/_datetime_encoder.py

```diff
@@ -180,95 +531,145 @@ def fit(self, X: ArrayLike, y=None) -> "DatetimeEncoder":
-                    self.features_per_column_[i].append("total_time")
-            else:
-                for feature in TIME_LEVELS:
-                    if np.nanstd(self._extract_from_date(X[:, i], feature)) > 0:
-                        if TIME_LEVELS.index(feature) <= TIME_LEVELS.index(
-                            self.extract_until
-                        ):
-                            self.features_per_column_[i].append(feature)
-                        # we add a total_time feature, which contains the full
-                        # time to epoch, if there is at least one
-                        # feature that has not been extracted and is not constant
-                        if TIME_LEVELS.index(feature) > TIME_LEVELS.index(
-                            self.extract_until
-                        ):
-                            self.features_per_column_[i].append("total_time")
-                            break
-                # Add day of the week feature if needed
-                if (
-                    self.add_day_of_the_week
-                    and np.nanstd(self._extract_from_date(X[:, i], "dayofweek")) > 0
-                ):
-                    self.features_per_column_[i].append("dayofweek")
-
-        self.n_features_in_ = X.shape[1]
-        self.n_features_out_ = len(
-            np.concatenate(list(self.features_per_column_.values()))
+        if self.resolution not in TIME_LEVELS and self.resolution is not None:
+            raise ValueError(
+                f"'resolution' options are {TIME_LEVELS}, got {self.resolution!r}."
+            )
+
+        errors_options = ["coerce", "raise"]
+        if self.errors not in errors_options:
+            raise ValueError(
+                f"'errors' options are {errors_options!r}, got {self.errors!r}."
+            )
+
+        self._check_feature_names(X, reset=True)
+        self._check_n_features(X, reset=True)
+        X = check_array(
+            X, ensure_2d=True, force_all_finite=False, dtype=None, copy=False
         )
 
+        self._select_datetime_cols(X)
+
         return self
 
-    def transform(self, X: ArrayLike, y=None) -> NDArray:
+    def _select_datetime_cols(self, X):
+        """Select datetime-parsable columns and generate the list of
+        datetime feature to extract.
+
+        If the input only contains dates (and no datetimes), only the features
+        ["year", "month", "day"] will be filtered with resolution.
+
+        Parameters
+        ----------
+        X : array-like of shape ``(n_samples, n_features)``
+        """
+        if self.resolution is None:
+            levels = []
+        else:
+            idx_level = TIME_LEVELS.index(self.resolution)
+            levels = TIME_LEVELS[: idx_level + 1]
+
+        X_split = np.hsplit(X, X.shape[1])
+        self.column_indices_, self.index_to_format_ = _get_datetime_column_indices(
+            X_split
+        )
+        del X_split
```

Comment:
why do we need to delete it explicitly?

Reply:
It's not as useful as I thought, let's remove it